oxide-auth/src/endpoint/mod.rs

```diff
@@ -178,8 +178,8 @@ enum InnerTemplate<'a> {
 ///
 /// [`OwnerSolicitor`]: trait.OwnerSolicitor.html
 pub struct Solicitation<'flow> {
-    pub(crate) grant: Cow<'flow, PreGrant>,
-    pub(crate) state: Option<Cow<'flow, str>>,
+    pub grant: Cow<'flow, PreGrant>,
+    pub state: Option<Cow<'flow, str>>,
```

Comment:
Here you can use `new` and `with_state` instead of making attributes public.

Reply:
Was able to use `Solicitation::new` instead